kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/EngineServiceDiscovery.scala

```diff
@@ -31,21 +31,27 @@ class EngineServiceDiscovery(
     fe: FrontendService) extends ServiceDiscovery("EngineServiceDiscovery", fe) {
 
   override def stop(): Unit = synchronized {
-    discoveryClient.deregisterService()
-    conf.get(ENGINE_SHARE_LEVEL) match {
-      // For connection level, we should clean up the namespace in zk in case the disk stress.
-      case "CONNECTION" =>
-        try {
-          if (discoveryClient.postDeregisterService) {
-            info("Clean up discovery service due to this is connection share level.")
+    if (!isServerLost.get()) {
+      discoveryClient.deregisterService()
+      conf.get(ENGINE_SHARE_LEVEL) match {
+        // For connection level, we should clean up the namespace in zk in case the disk stress.
+        case "CONNECTION" =>
+          try {
+            if (discoveryClient.postDeregisterService) {
```

Comment:
Shall we still clean the zk namespace even the Kyuubi server lost ?

Here we should skip `discoveryClient.deregisterService()` since the server node with its sequential number is not existed, but the namspace is alived.

Reply:
we are not handling kyuubi server lost at all

Reply:
I have made the pr clear to the zk server not kyuubi. your suggestion will introduce extra connection timeout+session timeout for this unnecessary operation